### PR TITLE
Update protocol upgrade deploy scripts

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

Task: [OS-522](https://aragonassociation.atlassian.net/browse/OS-522)

## Type of change

- deploy scripts

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] ~~I have updated the `CHANGELOG.md` file in the root folder.~~
- [x] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [x] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-522]: https://aragonassociation.atlassian.net/browse/OS-522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: UPDATE_CHECKLIST.md [hunk 1]
```diff
@@ -107,10 +108,19 @@ Nothing to do.
...
-- [ ] Verify that Release 1 Build 2 of the multisig plugin has been created
-- [ ] Verify that Release 1 Build 2 of the token voting plugin has been created
-- [ ] Verify that Release 1 Build 2 of the addresslist voting plugin has been created
+- [ ] Verify that the `DAO` base contract in the `DAOFactory`
+- [ ] Verify that the managing DAO implementation has been updated to the new implementation
+- [ ] Verify that the managing DAO is reinitialized to `_initialized = 2`
+- [ ] Verify that the old `DAOFactory` has no permissions on the `DAORegistry`
+- [ ] Verify that the new `DAOFactory` has the `REGISTER_DAO_PERMISSION_ID` permission on the `DAORegistry`
+- [ ] Verify that the `PluginRepo` base contract in the PluginRepoFactory has been updated
+- [ ] Verify that all `PluginRepo`s controlled by the managing DAO have been updated to the new implementationare AND still initialized with `_initialized = 1`
```
[the remark this answers is not in the document]
Comment (mathewmeconry) [reply UPDATE_CHECKLIST.md:117]: also from my point of view the `AND` doesn't need to be all caps. Somebody deploying stuff should be thorough enough to follow the list to the teeth without such things

**Review by mathewmeconry — CHANGES_REQUESTED** [review]
are the new scripts and the changes tested against a fork to make sure they work?

File: UPDATE_CHECKLIST.md [hunk 2]
```diff
@@ -107,10 +108,19 @@ Nothing to do.
...
 Wait until the managing DAO has made the necessary changes and then:
 
-- [ ] Verify that the managingDAO implementation has been updated
-- [ ] Verify that the managingDAO is reinitialized to `_initialized = 2`
-- [ ] Verify that the old DAOFactory has no permissions on the DAORegistry
-- [ ] Verify that the new DAOFactory has the `REGISTER_DAO_PERMISSION_ID` permission on the DAORegistry
-- [ ] Verify that Release 1 Build 2 of the multisig plugin has been created
-- [ ] Verify that Release 1 Build 2 of the token voting plugin has been created
-- [ ] Verify that Release 1 Build 2 of the addresslist voting plugin has been created
+- [ ] Verify that the `DAO` base contract in the `DAOFactory`
```
[the remark this answers is not in the document]
Comment (Michael-A-Heuer) [reply UPDATE_CHECKLIST.md]: Fixed in 48b2a7976c317eef6c0f6ece4fc1909d9f3636f4

File: UPDATE_CHECKLIST.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Michael-A-Heuer) [reply UPDATE_CHECKLIST.md]: Fixed in 48b2a7976c317eef6c0f6ece4fc1909d9f3636f4

File: packages/contracts/deploy/update/to_v1.3.0/90_ManagingDAO.ts [hunk 3]
```diff
@@ -5,32 +5,32 @@ import {getContractAddress} from '../../helpers';
...
     managingDAOAddress,
     hre.ethers.provider
   );
-  const upgradeTX = await daoInterface.populateTransaction.upgradeToAndCall(
-    daoImplementation,
-    daoInterface.interface.encodeFunctionData('initializeFrom', [[1, 0, 0], ''])
+  const upgradeTX = await managingDAO.populateTransaction.upgradeToAndCall(
+    newDaoImplementation,
+    managingDAO.interface.encodeFunctionData('initializeFrom', [[1, 0, 0], []])
   );
```
Comment (Rekard0) [inline packages/contracts/deploy/update/to_v1.3.0/90_ManagingDAO.ts:24]: i wonder if this DAO also need to upgrade itself,
why not do it later as well, perhaps via UI

**Review by Rekard0 — DISMISSED** [review]
LGTM

**Comment by Michael-A-Heuer** [comment]
> are the new scripts and the changes tested against a fork to make sure they work?

No, except for the tests in `packages/contracts/test/deploy/updateTo1_3_0.ts` no simulation of the update with tenderly or hardhat hardfork was done.

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**